executors/rust/src/decimalfmt.rs

```diff
@@ -35,17 +30,17 @@ pub fn run_numberformat_test(json_obj: &Value) {
     options.grouping_strategy = options::GroupingStrategy::Min2;
 
     let fdf = FixedDecimalFormatter::try_new_with_buffer_provider(
-        &provider, &data_locale, options)
+        &provider, &langid.into(), options)
```

Comment:
I prefer using the previous `DestType::from(x)` rather than `x.into()` because it's explicit to the reader about which type you get back.

Reply:
I agree in the general case, I find `.into()`s confusing. However in the context of `DataLocale` in ICU4X, we usually (always?) use `.into()` in our docs, because `DataLocale` is not meant to be a super public type. It required importing the `icu_provider` crates, which for most users (such as here), is an implementation detail.

Reply:
Okay, in that case, LGTM